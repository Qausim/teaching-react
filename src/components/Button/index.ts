export * from './RedButton';
export * from './BaseButton';
export * from './GreenButton';
