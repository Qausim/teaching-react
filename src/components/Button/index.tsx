import { ReactNode } from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.module.css';


type ButtonProps = {
  onClick?: () => void;
  className: string;
  children: ReactNode;
};


const Button = ({ children, className, onClick }: ButtonProps) => {
  return (
    <button className={className} onClick={onClick}>
      {children}
    </button>
  );
};

type PrimaryButtonProps = Omit<ButtonProps, 'className'>;

const PrimaryButton = ({ children, ...rest }: PrimaryButtonProps) => {
  return (
    <Button className={styles.primary} {...rest}>
      {children}
    </Button>
  );
}

type LinkButtonProps = Omit<ButtonProps, 'className'> & {
  href: string;
}

const LinkButton = ({ children, href }: LinkButtonProps) => {
  return (
    <Link to={href} className={styles.primary}>
      {children}
    </Link>
  );
}

Button.Primary = PrimaryButton;
Button.Link = LinkButton;

export default Button;
