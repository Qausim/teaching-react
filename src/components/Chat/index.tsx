import { MessageType } from "../../data/static-data";
import { useAppDispatch, useAppSelector } from "../../store";
import { deleteMessage } from "../../store/reducers/messages.reducer";
// import { deleteMessage } from "../../store/actions/messages.actions";

type ChatProps = {
  message: MessageType;
};

const Chat = ({ message }: ChatProps) => {
  const { text, is_user_msg } = message;
  const dispatch = useAppDispatch();
  const activeUserId = useAppSelector((state) => state.chatState.activeUserId);

  const handleDelete = () => {
    dispatch(deleteMessage({ id: message.message_id, userId: activeUserId!}));
  }

  return (
    <div
      className={`Chat ${is_user_msg ? "is-user-msg" : ""}`}
    >
      {is_user_msg && <button onClick={handleDelete} className="del-btn">&times;</button>}
      {text}
    </div>
  );
}

export default Chat;
