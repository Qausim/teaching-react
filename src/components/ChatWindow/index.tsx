import ChatHeader from "../../containers/ChatHeader";
import Chats from "../../containers/Chats";
import MessageInput from "../../containers/MessageInput";
import './ChatWindow.css'

const ChatWindow = () => {
  return (
    <div className="ChatWindow">
      <ChatHeader />
      <Chats />
      <MessageInput />
    </div>
  )
};

export default ChatWindow;
