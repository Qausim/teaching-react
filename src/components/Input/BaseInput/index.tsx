import React from 'react';
import styled from 'styled-components';


const Input = styled.input`
  & {
    padding: 4px;
    border-radius: 4px;
  }
`;

type InputProps = {
  [key: string]: any
}

const BaseInput = (props: InputProps) => {
  return (
    <Input {...props}/>
  )
};

export default BaseInput;
