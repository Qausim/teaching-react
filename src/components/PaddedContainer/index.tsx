import { ReactNode } from "react";
import styles from './styles.module.css';

const PaddedContainer = ({ children }: { children: ReactNode; }) => {
  return (
    <div className={styles.padded_container}>
      {children}
    </div>
  )
};

export default PaddedContainer;
