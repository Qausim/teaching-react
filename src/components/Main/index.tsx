import ChatWindow from "../ChatWindow";
import Empty from "../../containers/Empty";
import store, { useAppSelector } from "../../store";
import './Main.css';


const Main = () => {
  const activeUserId = useAppSelector((state) => state.chatState.activeUserId);
  return (
    <main className="Main">
      {
        activeUserId ? <ChatWindow /> : <Empty />
      }
    </main>
  );
};

export default Main;
