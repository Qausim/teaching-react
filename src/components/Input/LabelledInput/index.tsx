import React from 'react';
import styled, { css } from 'styled-components';
import BaseInput from '../BaseInput';


interface WrapperProps {
  width?: string | number;
}

const Wrapper = styled.div<WrapperProps>`
  /* & {
    width: ${props => props.width || '100%'};
  } */

  ${(props) => props.width && css`
    & {
      width: ${props.width};
      margin: 0 auto;
    }

    input {
      width: 100%;
    }
  `}

  .helper-text {
    small, span {
      display: block;
    }
  }
`;


interface Props extends React.HTMLProps<HTMLInputElement> {}

export const LabelledInput = ({id, width, label, ...rest}: Props) => {
  return (
    <Wrapper width={width}>
      <label htmlFor={id}>{label}</label>
      <BaseInput {...{id, ...rest}}/>
      <p className='helper-text'>
        <small>Whatever!!!</small>
        <span>Whatever you too. Petty!!!</span>
      </p>
    </Wrapper>
  );
}
