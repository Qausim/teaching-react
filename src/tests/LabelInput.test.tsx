import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { LabelledInput } from "../components/Input";
import axios from 'axios';


jest.mock('axios');

const wrapper = {
  doSomething() {
    console.log('INSIDE doSomething...');
  }
}

const otherDo = () => {
  wrapper.doSomething();
}


describe('otherDo', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should call wrapper.doSomething', () => {
    const doSomethingSpy = jest.spyOn(wrapper, 'doSomething');
    otherDo();

    expect(doSomethingSpy).toHaveBeenCalled();
  });

  it('should mock axios', async () => {
    // @ts-ignore
    axios.get.mockResolvedValue({ data: { id: 1, name: 'Qausim' } });

    const data = await axios.get('http://localhost:9000');

    expect(data).toBeDefined();
  });
});

describe('LabelInput', () => {
  it('should render input element with label', () => {
    const mockChangeHandler = jest.fn();
    const mockFocusHandler = jest.fn();
    const mockBlurHandler = jest.fn();

    render(
      <LabelledInput
        id="my-input"
        label="My Input"
        value="something"
        onChange={mockChangeHandler}
        onFocus={mockFocusHandler}
        onBlur={mockBlurHandler}
      />
    );

    const input = screen.getByDisplayValue("something");
    const label = screen.getByText(/my input/i);
    expect(input).toHaveAttribute("id", "my-input");
    expect(label).toHaveAttribute("for", "my-input");
    fireEvent.input(input, { target: { value: "whatever" } });
    expect(mockChangeHandler).toBeCalledTimes(1);
    input.focus();
    expect(mockFocusHandler).toBeCalled();
    input.blur();
    expect(mockBlurHandler).toBeCalled();
  });

  it('should match snapshot', () => {
    const mockChangeHandler = jest.fn();

    expect(render(
      <LabelledInput
        id="my-input"
        label="My Input"
        value="something"
        onChange={mockChangeHandler}
      />
    )).toMatchSnapshot();
  });
});
