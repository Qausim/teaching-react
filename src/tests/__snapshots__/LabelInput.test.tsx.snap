// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelInput should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsnTZi isZefa"
      >
        <label
          for="my-input"
        >
          My Input
        </label>
        <input
          class="sc-bczRLJ dACYAo"
          id="my-input"
          value="something"
        />
        <p
          class="helper-text"
        >
          <small>
            Whatever!!!
          </small>
          <span>
            Whatever you too. Petty!!!
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsnTZi isZefa"
    >
      <label
        for="my-input"
      >
        My Input
      </label>
      <input
        class="sc-bczRLJ dACYAo"
        id="my-input"
        value="something"
      />
      <p
        class="helper-text"
      >
        <small>
          Whatever!!!
        </small>
        <span>
          Whatever you too. Petty!!!
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
